reporting/src/main/java/org/wso2/testgrid/reporting/ChartGenerator.java

```diff
@@ -0,0 +1,152 @@
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Map;
+import javax.imageio.ImageIO;
+
+/**
+ * This class is responsible for generating the necessary charts for the email report.
+ *
+ * @since 1.0.0
+ */
+public class ChartGenerator {
+
+    private static final Logger logger = LoggerFactory.getLogger(ChartGenerator.class);
+    private static  final String summaryChartFileName = "summary.png";
+    private static final String historyChartFileName = "history.png";
+    private String chartGenLocation;
+
+    public ChartGenerator(String chartGenLocation) {
+        new JFXPanel();
+        this.chartGenLocation = chartGenLocation;
+    }
+
+    /**
+     * Generates a pie chart with the summary test results of the current build.
+     *
+     * @param passedCount  passed test count
+     * @param failedCount  failed test count
+     * @param skippedCount skipped test count
+     * @throws IOException if the chart cannot be written into a file
+     */
+    public void generateSummaryChart(int passedCount, int failedCount, int skippedCount) {
+
+        ObservableList<PieChart.Data> pieChartData =
+                FXCollections.observableArrayList(
+                        new PieChart.Data("Failed", failedCount),
+                        new PieChart.Data("Skipped", skippedCount),
+                        new PieChart.Data("Passed", passedCount));
+        final PieChart chart = new PieChart(pieChartData);
+        chart.setAnimated(false);
+        chart.setTitle("Build Failure Summary by Test Plans");
+        genChart(chart, 500, 500, summaryChartFileName);
+    }
+
+    /**
+     * Generates the history chart with the summary of test executions.
+     *
+     * @param dataSet input data-set for the chart
+     */
+    public void generateResultHistoryChart(Map<String, BuildExecutionSummary> dataSet) {
+
+        final CategoryAxis xAxis = new CategoryAxis();
+        final NumberAxis yAxis = new NumberAxis();
+        final StackedBarChart<String, Number> stackedBarChart = new StackedBarChart<>(xAxis, yAxis);
+        // This represents the series of values e.g: Failed, skipped, Passed
+        final XYChart.Series<String, Number>[] seriesSet = new XYChart.Series[]{new XYChart.Series<>(),
+                new XYChart.Series<>(), new XYChart.Series<>()};
+
+        xAxis.setCategories(FXCollections.<String>observableArrayList(dataSet.keySet()));
+        // Disabling animation
+        xAxis.setAnimated(false);
+        yAxis.setAnimated(false);
+        stackedBarChart.setAnimated(false);
+        // Set Axis Names
+        xAxis.setLabel("Build Timestamp");
+        yAxis.setLabel("Number of Infra Combinations");
+
+        // Setting series names
+        seriesSet[0].setName("Build Failed Combinations");
+        seriesSet[1].setName("Build Passed Combinations");
+        seriesSet[2].setName("Infra Failed Combinations");
+        // Setting space between the bars
+        stackedBarChart.setCategoryGap(50);
```

Comment:
From where did this magic number come from? :)

Reply:
This is an aesthetic improvement, so just added a number which looked nice :)